blockmanager.go

```diff
@@ -659,33 +659,22 @@ func (b *blockManager) startSync(peers *list.List) {
 // syncing from.
 func (b *blockManager) isSyncCandidate(sp *serverPeer) bool {
 	// The peer is not a candidate for sync if it's not a full node.
-	if sp.Services()&wire.SFNodeNetwork != wire.SFNodeNetwork {
-		return false
-	}
-
-	// Candidate if all checks passed.
-	return true
+	return sp.Services()&wire.SFNodeNetwork == wire.SFNodeNetwork
 }
 
 // syncMiningStateAfterSync polls the blockMananger for the current sync
 // state; if the mananger is synced, it executes a call to the peer to
 // sync the mining state to the network.
 func (b *blockManager) syncMiningStateAfterSync(sp *serverPeer) {
 	go func() {
-		ticker := time.NewTicker(time.Second * 3)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ticker.C:
-				if !sp.Connected() {
-					return
-				}
-				if b.IsCurrent() {
-					msg := wire.NewMsgGetMiningState()
-					sp.QueueMessage(msg, nil)
-					return
-				}
-			}
+		time.Sleep(3 * time.Second)
```

Comment:
this is completely different behavior, you must keep the loop.  do this: `for range <-ticker.C {`